wagtail/api/v2/tests/test_pages.py

```diff
@@ -81,6 +84,48 @@ def test_private_pages_dont_appear_in_list(self):
         content = json.loads(response.content.decode('UTF-8'))
         self.assertEqual(content['meta']['total_count'], new_total_count)
 
+    def test_private_pages_with_user_login(self):
+        client = APIClient()
```

Comment:
Any reason to not use ``self.client`` here?

Reply:
For some reason when I used `self.client` and `self.login` the user wasn't logged in the views and always returned `anonymous` user type